.github/workflows/java-roll-version-and-release.yaml

```diff
@@ -0,0 +1,110 @@
+        run: |
+          echo ${{steps.major.outputs.major}}.${{steps.minor.outputs.minor}}.${{steps.patch.outputs.patch}}
+  update-version-commit:
+    needs: extract
+    env:
+      NEW_VERSION: ${{ needs.extract.outputs.major }}.${{ needs.extract.outputs.minor }}.${{ needs.extract.outputs.patch }}
+    runs-on:
+      - self-hosted
+      - java
+    steps:
+      - run: echo "NEW_VERSION=$NEW_VERSION"
+      - uses: actions/checkout@v4
+      - name: Setup git author information
+        run: |
+          git config user.name RCSBSystem
+          git config user.email it@rcsb.org
+      - run: mvn --version
+      - name: Set version
+        run: mvn versions:set -DnewVersion=$NEW_VERSION --no-transfer-progress
+      - name: Commit and tag new version (release commit)
+        run: |
+          git add $(find . -name "pom.xml")
+          git commit -m "[actions] Updated version in pom.xml to $NEW_VERSION"
+          git tag $NEW_VERSION
+      - name: Push back to GitHub
+        # Note we must push or otherwise next job won't have the commit (every job starts from scratch)
+        run: |
+          git push --all
+          git push origin $NEW_VERSION
+
+  build_jib_production:
+    name: "Build docker image via jib with tags: production,${{ needs.extract.outputs.major }}.${{ needs.extract.outputs.minor }}.${{ needs.extract.outputs.patch }}"
+    needs: update-version-commit
+    uses: ./.github/workflows/run_mvn.yaml
+    with:
+      script: package jib:build -Djib.to.tags=production,${{ needs.extract.outputs.major }}.${{ needs.extract.outputs.minor }}.${{ needs.extract.outputs.patch }} -Djib.target.namespace=rcsb --no-transfer-progress -Dmaven.test.skip=true -Djib.httpTimeout=0
+
+  # TODO remove once we are out of OpenStack and build-locker artifacts are not needed anymore
+  call-deploy-to-build-locker:
+    needs: update-version-commit
+    uses: ./.github/workflows/deploy-to-buildlocker.yaml
+    with:
+      distribution: "${{ inputs.distribution }}"
+      type: "${{ inputs.type }}"
+
+  next-dev-cycle-commit:
+    needs:
+      - call-deploy-to-build-locker
+      - build_jib_production
+    env:
+      NEW_VERSION: ${{ needs.extract.outputs.major }}.${{ needs.extract.outputs.minor }}.${{ needs.extract.outputs.patch }}
+    runs-on:
+      - self-hosted
+      - java
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup git author information
+        run: |
+          git config user.name RCSBSystem
+          git config user.email it@rcsb.org
+      - name: Increment version
+        run: |
+          NEXT_PATCH=$(mvn build-helper:parse-version help:evaluate -Dexpression=parsedVersion.nextIncrementalVersion -q -DforceStdout --no-transfer-progress)
+          mvn versions:set -DnewVersion="${{ inputs.major }}.${{ inputs.minor }}.$NEXT_PATCH-SNAPSHOT" -DallowSnapshots --no-transfer-progress
+      - name: Commit new version for next dev cycle
+        run: |
+          git add $(find . -name "pom.xml")
```

Comment:
This will behave weirdly if there's more than one `pom.xml`. Can it just be `git add pom.xml`?

Reply:
In multimodule projects you have several `pom.xml`s (in subdirs). I guess that's why Igor added it.